test/jesse_schema_validator_tests.erl

```diff
@@ -221,6 +221,73 @@ schema_unsupported_test() ->
               , jesse_schema_validator:validate(UnsupportedSchema, Json, [])
               ).
 
+data_invalid_one_of_test() ->
+  IntegerSchema = {[{<<"type">>, <<"integer">>}]},
+  StringSchema  = {[{<<"type">>, <<"string">>}]},
+  ObjectSchema  = {[ {<<"type">>, <<"object">>}
+                   , { <<"properties">>
+                     , [ {<<"name">>, StringSchema}
+                       , {<<"age">>, IntegerSchema}
+                       ]
+                     }
+                   , {<<"additionalProperties">>, false}
+                   ]},
+
+  Schema = {[ {<<"$schema">>, <<"http://json-schema.org/draft-04/schema#">>}
+            , {<<"oneOf">>, [IntegerSchema, StringSchema, ObjectSchema]}
+            ]},
+
+  Json = [ {<<"name">>, 42}
+         , {<<"age">>, <<"John">>}
+         ],
+
+  ?assertThrow(
+     [ { data_invalid
+       , Schema
+       , { not_one_schema_valid
+         , [ {data_invalid, IntegerSchema, wrong_type, Json, []}
+           , {data_invalid, StringSchema, wrong_type, Json, []}
+           , {data_invalid, StringSchema, wrong_type, 42, [<<"name">>]}
+           ]
+         }
+       , Json
+       , []
+       }],
+     jesse_schema_validator:validate(Schema, Json, [])
+    ).
+
+data_invalid_any_of_test() ->
+  IntegerSchema = {[{<<"type">>, <<"integer">>}]},
+  StringSchema  = {[{<<"type">>, <<"string">>}]},
+  ObjectSchema  = {[ {<<"type">>, <<"object">>}
+                   , { <<"properties">>
+                     , [ {<<"name">>, StringSchema}
+                       , {<<"age">>, IntegerSchema}
+                       ]
+                     }
+                   , {<<"additionalProperties">>, false}
+                   ]},
+
+  Schema = {[ {<<"$schema">>, <<"http://json-schema.org/draft-04/schema#">>}
+            , {<<"anyOf">>, [IntegerSchema, StringSchema, ObjectSchema]}
+            ]},
+
+  Json = [ {<<"name">>, 42}
+         , {<<"age">>, <<"John">>}
+         ],
+
+  ?assertThrow(
+     [ { data_invalid
+       , Schema
+       , { any_schemas_not_valid
+         , [{data_invalid, IntegerSchema, wrong_type, Json, []}]
```

Comment:
shouldn't it return 3 sub-errors?

Reply:
ah, this PR differs from #91 only for `oneOf`... But why not return full error list for `anyOf` and `allOf`?

Reply:
The reason why the same is not applied to `allOf` and `anyOf` is because the nature of those restrictions is different, and it is a lot easier to figure out what the problem is in case of error.

If we return all errors for `anyOf` it will return an error for most of its sub-schemas, most of the time. 

Maybe it does make sense to return all errors for the `allOf` case.